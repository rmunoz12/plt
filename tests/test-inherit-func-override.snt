@setup {

group A(Object) {
    int x;

    func group A __init__(int x) {
        this.x = x;
        return this;
    }

    func int test() {
        return this.x + 5;
    }

};


group B(A) {
    int y;

    func int test() {
        this.y = 10;
        return this.y;
    }
};

group C(A) {

};

group D(B) {
    func group D __init__(int x, int y) {
        this.x = x;
        this.y = y;
        return this;
    }
};

group E(D) {
    func bool test(int x) {
        this.x = x;
        if (this.x > 0) {
            return (True);
        } else {
            return (False);
        }
    }
};

group A obj;
group B obj2;
group C obj3;
group D obj4;
group E obj5;

}

@turns {

func void begin() {
    int y;

    obj.__init__(1);
    obj2.__init__(2);
    obj3.__init__(3);
    obj4.__init__(20, 30);
    obj5.__init__(0, 1);

    print(obj.x); print("\n");
    print(obj2.x); print("\n");
    print(obj3.x); print("\n");
    print(obj4.x); print("\n");
    print(obj4.y); print("\n");

    print(obj.test()); print("\n");
    print(obj2.test()); print("\n");
    print(obj3.test()); print("\n");
    print(obj4.test()); print("\n");

    print(obj4.y); print("\n");

    print(obj5.test()); print("\n");
    print(obj5.test(5)); print("\n");


    end;
}

}

