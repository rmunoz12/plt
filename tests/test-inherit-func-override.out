1
2
3
20
30
6
10
8
10
10
10
true
